File: src/org/jetbrains/agda/core/CExpression.java
package org.jetbrains.agda.core;

public class CExpression {
    private CExpression myParent = null;

    public CExpression() {

    }

    public CExpression getParent() {
        return myParent;
    }

    protected void attach(CExpression expression) {
        expression.myParent = this;
    }
}

File: src/org/jetbrains/agda/core/expression/CExpression.kt
package org.jetbrains.agda.core

/**
 * @author Evgeny.Kurbatsky
 */
public abstract class CExpression() {
    private var myParent : CExpression? = null
    private var myType : CExpression? = null

    public fun getParent() : CExpression? {
        return myParent
    }
    public fun getType() : CExpression? {
        return myType
    }
    protected fun attach(expression : CExpression) : Unit {
        expression.myParent = this
    }
    public fun setType(aType : CExpression?) : Unit {
        this.myType = aType
    }

}
